### PR TITLE
frontend-plugin-api: refactor createThemeExtension to accept both theme and name

### PR DESCRIPTION
**Pull request by Rugvip**
## Hey, I just made a Pull Request!

Work towards #19545, related to https://github.com/backstage/backstage/pull/21481#discussion_r1406088848. This is the same change as was applied to API extensions in #21545.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: packages/frontend-plugin-api/src/extensions/createThemeExtension.ts [hunk 1]
```diff
@@ -18,11 +18,15 @@ import { createExtension, coreExtensionData } from '../wiring';
 import { AppTheme } from '@backstage/core-plugin-api';
 
 /** @public */
-export function createThemeExtension(theme: AppTheme) {
+export function createThemeExtension(options: {
+  name?: string;
+  theme: AppTheme;
```
[the remark this answers is not in the document]
Comment (freben) [reply packages/frontend-plugin-api/src/extensions/createThemeExtension.ts:23]: you mean just like it was before the change?

[the remark this answers is not in the document]
Comment (tudi2d) [reply packages/frontend-plugin-api/src/extensions/createThemeExtension.ts:23]: ah true - yes, just like it was before

[the remark this answers is not in the document]
Comment (Rugvip) [reply packages/frontend-plugin-api/src/extensions/createThemeExtension.ts:23]: I've noted this down as the first Q for polish in #19545. There are plenty of places where we may want to get rid of one layer of types. Here, APIs, translations, probably more. For now in this PR I think the question is whether we want to merge `AppTheme` into options. I could switch it to just just so we have some different usages for future discussion.

[the remark this answers is not in the document]
Comment (Rugvip) [reply packages/frontend-plugin-api/src/extensions/createThemeExtension.ts:23]: On second thought, the `AppTheme` has an `id`, which makes it pretty confusing to have at the top level I think. Keeping it as is for now.

[the remark this answers is not in the document]
Comment (tudi2d) [reply packages/frontend-plugin-api/src/extensions/createThemeExtension.ts:23]: so I don't see a problem with merging `AppTheme` into `options` for `createThemeExtension`. An escape could be to have a second parameter so smth. like: `createThemeExtension(theme: AppTheme, options?: {name: string})`. This way you could prevent the confusion with `id`. 

Agree to @freben though - why is `name` not required? :)

**Pull request closed without merging**